coops-ui/src/components/AnnualReport/RegisteredOfficeAddress.vue

```diff
@@ -285,38 +304,42 @@ export default class RegisteredOfficeAddress extends Vue {
   private resetAddress (): void {
     this.deliveryAddress = { ...this.deliveryAddressOriginal }
     this.mailingAddress = { ...this.mailingAddressOriginal }
+    this.emitAddresses()
     this.emitModified()
   }
 
   /**
    * Loads the office addresses from an API call.
-   *
-   * @param legalEntityNumber the identifier for the legal entity whose office addresses are fetched.
    */
   private loadAddressesFromApi (legalEntityNumber: string): void {
     if (legalEntityNumber) {
       const url = legalEntityNumber + '/addresses'
       axios.get(url)
         .then(response => {
           if (response && response.data) {
-            if (response.data.deliveryAddress) {
-              this.deliveryAddressOriginal = { ...response.data.deliveryAddress }
-              this.deliveryAddress = { ...response.data.deliveryAddress }
+            const deliveryAddress = response.data.deliveryAddress
+            if (deliveryAddress) {
+              this.deliveryAddressOriginal = { ...deliveryAddress }
+              // If parent page loaded draft before this API call, don't overwrite draft delivery address.
```

Comment:
Hopefully it won't be long until we can move this API call out of the class, and into the parent. It will make things so much simpler.

Reply:
Hmm, yes and no. This component needs to fetch the "current" addresses, either as a starting point for a new filing or a comparison for a draft filing. To get around this, we would have to pass in 2 addresses (the draft address and the current address). I don't know which is cleaner...